test/module/irohad/consensus/yac/mocks/crypto_provider.hpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gmock/gmock.h>
+
+#include "consensus/yac/yac_crypto_provider.hpp"
+#include "cryptography/crypto_provider/crypto_defaults.hpp"
+
+#include "module/shared_model/interface_mocks.hpp"
+
+#ifndef IROHA_CRYPTO_PROVIDER_HPP
```

Comment:
This header file name seems too generic, since there could be another `CryptoProvider` in the project, which will have the same filename. The class in this file is called `MockYacCryptoProvider`, and the folder structure is not reflected in the header guard. Also, we have (mostly) followed the policy, which suggests to have a single class in a header file, and the file has the same name as the class.
I would suggest to call the file `mock_yac_crypto_provider.hpp`, and put it directly in ` test/module/irohad/consensus/yac` directory.
Same applies to other mocks in this pull request.